include/seqan3/alphabet/structure/structured_aa.hpp

```diff
@@ -0,0 +1,170 @@
+ * correpsond to numeric values in the size of the composition, while the character values
+ * are taken from the sequence alphabet and the structure annotation is taken from the structure
+ * alphabet.
+ *
+ * As with all `seqan3::cartesian_composition` s you may access the individual alphabet letters in
+ * regular c++ tuple notation, i.e. `get<0>(t)` and objects can be brace-initialized
+ * with the individual members.
+ *
+ * ~~~~~~~~~~~~~~~{.cpp}
+ *
+ * structured_aa<aa27, dssp9> l{aa27::W, dssp9::B};
+ * std::cout << int(to_rank(l)) << ' '
+ *           << int(to_rank(get<0>(l))) << ' '
+ *           << int(to_rank(get<1>(l))) << '\n';
+ * // 49 22 1
+ *
+ * std::cout << to_char(l) << ' '
+ *           << to_char(get<0>(l)) << ' '
+ *           << to_char(get<1>(l)) << '\n';
+ * // W W B
+ *
+ * // modify via structured bindings and references:
+ * auto & [ seq_l, structure_l ] = l;
+ * seq_l = aa27::V;
+ * std::cout << to_char(l) << '\n';
+ * // V
+ *
+ * ~~~~~~~~~~~~~~~
+ *
+ * This seqan3::cartesian_composition itself fulfills seqan3::alphabet_concept.
+ */
+
+template <typename sequence_alphabet_t = aa27, typename structure_alphabet_t = dssp9>
+requires alphabet_concept<sequence_alphabet_t> && alphabet_concept<structure_alphabet_t>
+struct structured_aa :
+    public cartesian_composition<structured_aa<sequence_alphabet_t, structure_alphabet_t>,
+                                 sequence_alphabet_t, structure_alphabet_t>
+{
+    //!\brief First template parameter as member type.
+    using sequence_alphabet_type = sequence_alphabet_t;
+    //!\brief Second template parameter as member type.
+    using structure_alphabet_type = structure_alphabet_t;
+
+    //!\brief Equals the char_type of sequence_alphabet_type.
+    using char_type = underlying_char_t<sequence_alphabet_type>;
+    //!\brief Equals the char_type of the structure_alphabet_type.
+    using str_char_type = underlying_char_t<structure_alphabet_type>;
+
+    /*!\name Write functions
+     * \{
+     */
+    //!\brief Directly assign the sequence character.
+    constexpr structured_aa & operator=(sequence_alphabet_type const l) noexcept
+    {
+        get<0>(*this) = l;
+        return *this;
+    }
+
+    //!\brief Directly assign the structure character.
+    constexpr structured_aa & operator=(structure_alphabet_type const l) noexcept
+    {
+        get<1>(*this) = l;
+        return *this;
+    }
+
+    //!\brief Assign from a nucleotide character. This modifies the internal sequence letter.
+    constexpr structured_aa & assign_char(char_type const c)
+    {
+        seqan3::assign_char(get<0>(*this), c);
+        return *this;
+    }
+
+    //!\brief Assign from a structure character. This modifies the internal structure letter.
+    constexpr structured_aa & assign_structure(str_char_type const c)
+    {
+        seqan3::assign_char(get<1>(*this), c);
+        return *this;
+    }
```

Comment:
Having this "top-level" assign and retrieve function for char is necessary to satisfy the alphabet concept, but I think the extra assign_structure confuses more than it helps. People should just use `get<1>` or `get<dssp9>` if they want to access the structure in the alphabet, especially since it uses the structure's char type which is even more confusing because `assign_structure(dssp9::H);` won't work.